tests/project/tsconf.tests.ts

```diff
@@ -0,0 +1,99 @@
+import ts, {CompilerOptions, ModuleKind, ParsedCommandLine, ScriptTarget} from 'typescript';
+import {rootDir, titled} from '../test-utils';
+import {join} from 'path';
+import {fork} from 'child_process';
+
+const tscBin = require.resolve('typescript/bin/tsc');
+
+describe('tsconfig', () => {
+    // Typescript config types
+    const tsConfigs: Record<string, CompilerOptions> = {
+        ES2019_ES2015_Chrome: {
+            target: ScriptTarget.ES2019, module: ModuleKind.ES2015,
+            lib: ['lib.es2015.d.ts'], types: ['chrome'],
+        },
+        ES2019_CommonJS_Jest_Node: {
+            target: ScriptTarget.ES2019, module: ModuleKind.CommonJS,
+            lib: [], types: ['jest', 'node'],
+        },
+        ES2019_CommonJS_Jest_Puppeteer: {
+            target: ScriptTarget.ES2019, module: ModuleKind.CommonJS,
+            lib: [], types: ['jest', 'puppeteer-core'],
+        },
+        ES2019_ES2015_Karma_Jasmine: {
+            target: ScriptTarget.ES2019, module: ModuleKind.ES2015,
+            lib: ['lib.es2015.d.ts'], types: ['chrome', 'jasmine'],
+        },
+    };
+
+    // Configs
+    const tests: Array<{ want: CompilerOptions; dir: string }> = [
+        {
+            dir: 'src',
+            want: tsConfigs.ES2019_ES2015_Chrome
+        },
+        {
+            dir: 'tests/browser',
+            want: tsConfigs.ES2019_CommonJS_Jest_Puppeteer,
+        },
+        {
+            dir: 'tests/config',
+            want: tsConfigs.ES2019_CommonJS_Jest_Node,
+        },
+        {
+            dir: 'tests/generators/utils',
+            want: tsConfigs.ES2019_CommonJS_Jest_Node,
+        },
+        {
+            dir: 'tests/inject',
+            want: tsConfigs.ES2019_ES2015_Karma_Jasmine,
+        },
+        {
+            dir: 'tests/project',
+            want: tsConfigs.ES2019_CommonJS_Jest_Node,
+        },
+        {
+            dir: 'tests/utils',
+            want: tsConfigs.ES2019_CommonJS_Jest_Node,
+        },
+    ];
+
+    it.each(tests.map(titled(({dir, want: {module, target}}) => {
+        return `./${dir}/ uses target ${ScriptTarget[target]} with module type ${ModuleKind[module]}`;
+    })))('$title', async ({want, dir}) => {
+        // Fire
+        const {options: got} = await showConfig(join(rootDir, dir));
+
+        expect({...got, lib: [], types: []}).toMatchObject({...want, lib: [], types: []});
```

Comment:
Instead of zeroing this, check the individual 2 fields(module + target)?

Reply:
The one thing we'd be sacrificing is the ability to add a key to the tests list and have it automatically tested. With the new change, you'd need to remember to add it to this list too.

Reply:
Done.